### PR TITLE
Fix a bug how missing values in ConfigurationItem are handled

### PR DESCRIPTION
**Pull request by davidanthoff**
Fixes https://github.com/julia-vscode/LanguageServer.jl/pull/416.

@non-Jedi does this work now for you?

### CODE REVIEW

File: src/protocol/configuration.jl [hunk 1]
```diff
@@ -234,11 +234,20 @@ mutable struct DidChangeConfiguration
...
-    scopeUri::Union{Nothing,String}    
+@json_read mutable struct ConfigurationItem
+    scopeUri::Union{Nothing,String}
     section::Union{Nothing,String}
 end
 
+function JSON.lower(a::ConfigurationItem)
+    d = Dict{String, Any}()
+
+    a.scopeUri!==nothing && d["scopeUri"] = a.scopeUri
```
Comment (non-Jedi) [inline src/protocol/configuration.jl]: Have to enclose the assignment in parentheses `&&` or `||` for this to be valid syntax:

```julia
julia> d = Dict()
Dict{Any,Any} with 0 entries

julia> true && d[1] = 5
ERROR: syntax: invalid assignment location "true && d[1]"
Stacktrace:
 [1] top-level scope at REPL[5]:1
 [2] run_backend(::REPL.REPLBackend) at /home/adam/.julia/packages/Revise/439di/src/Revise.jl:975
 [3] (::getfield(Revise, Symbol("##80#82")){REPL.REPLBackend})() at ./task.jl:268

julia> true && (d[1] = 5)
5
```

Comment (davidanthoff) [reply src/protocol/configuration.jl]: Oh man, how embarrassing :) New push that hopefully fixes that!

**Comment by non-Jedi** [comment]
I can't get this to precompile (d9772fb), so I can't test. Looks like this is the same reason Travis is failing:

```
julia> using LanguageServer
[ Info: Precompiling LanguageServer [2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7]
ERROR: LoadError: UndefVarError: Scope not defined
Stacktrace:
 [1] top-level scope at none:2
in expression starting at /home/adam/.julia/packages/StaticLint/FuPWe/src/StaticLint.jl:5
ERROR: LoadError: Failed to precompile StaticLint [b3cc710f-9c33-5bdb-a03d-a94903873e97] to /home/adam/.julia/compiled/v1.2/StaticLint/FgruE.ji.
Stacktrace:
 [1] top-level scope at none:2
in expression starting at /home/adam/.julia/packages/LanguageServer/x9ZqS/src/LanguageServer.jl:12
ERROR: Failed to precompile LanguageServer [2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7] to /home/adam/.julia/compiled/v1.2/LanguageServer/ite7n.ji.
```

**Comment by non-Jedi** [comment]
I am unable to test this due to https://github.com/julia-vscode/SymbolServer.jl/issues/57.

**Pull request merged**